wasm/src/main/scala/ir2wasm/WasmExpressionBuilder.scala

```diff
@@ -305,11 +306,111 @@ private class WasmExpressionBuilder private (
           IRTrees.ApplyStatically(t.flags, t.receiver, className, t.method, t.args)(t.tpe)(t.pos)
         )
 
+      case IRTypes.AnyType if t.method.name.isReflectiveProxy =>
+        genReflectiveCall(t)
```

Comment:
```suggestion
      case _ if t.method.name.isReflectiveProxy =>
        genReflectiveCall(t)
```
It's possible for the receiver to be a `ClassType` or an `ArrayType` and we still need to do the reflective call logic.